src/coinjoin/client.cpp

```diff
@@ -121,7 +121,7 @@ PeerMsgRet CCoinJoinClientQueueManager::ProcessDSQueue(const CNode& peer, CDataS
             LogPrint(BCLog::COINJOIN, "DSQUEUE -- new CoinJoin queue (%s) from masternode %s\n", dsq.ToString(),
                      dmn->pdmnState->addr.ToString());
 
-            ranges::any_of(m_walletman.raw(),
+            ranges::any_of(WITH_LOCK(m_walletman.cs_wallet_manager_map, return m_walletman.raw()),
```

Comment:
I'm pretty sure this is still a data race in these instances. You're acquiring the mutex, then returning the .raw(), then releasing the mutex, before using the .raw in the ranges::any_of. This may have made the issue not appear, but I think it could. You should probably guard the entire ranges::any_of (to also avoid thrashing the mutex)

Reply:
Resolved in latest push!